page/manager/AbstractPageManager.js

```diff
@@ -672,11 +672,11 @@ export default class AbstractPageManager extends PageManager {
    * @returns {Promise<any>}
    */
   async _runPreManageHandlers(nextManagedPage, action) {
-    if (!this._handlerRegistry) {
+    if (!this._pageHandlerRegistry) {
```

Comment:
Why do you check `this._pageHandlerRegistry`? Because PageHandlerRegistry is regular dependency.